src/client/views/cube-view/cube-view.tsx

```diff
@@ -229,11 +231,13 @@ export class CubeView extends React.Component<CubeViewProps, CubeViewState> {
     this.setState({ updatingMaxTime: true });
 
     DataCube.queryMaxTime(dataCube)
-      .then(updatedMaxTime => {
+      .then(maxTime => {
         if (!this.mounted) return;
+        if (!dataCube.refreshRule.isRealtime() && maxTime.getTime() === timekeeper.getTime(dataCube.name).getTime()) return;
         this.setState({
-          timekeeper: timekeeper.updateTime(dataCube.name, updatedMaxTime),
-          updatingMaxTime: false
+          timekeeper: timekeeper.updateTime(dataCube.name, maxTime),
+          updatingMaxTime: false,
+          requestTimestamp: (new Date()).getTime()
```

Comment:
It's only place where this is set - maybe we could just use incremented counter?

Reply:
Keep timestamp IMHO